### PR TITLE
Adding Origin-Agent-Cluster

### PR DESCRIPTION
**Pull request by salesh**
Related to https://github.com/helmetjs/helmet/issues/275

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## 4.3.2 - 2021-01-17
```
Comment (EvanHahn) [inline CHANGELOG.md:3]: Could you change this heading to "unreleased"?

File: README.md [hunk 2]
```diff
@@ -45,6 +45,7 @@ app.use(helmet.noSniff());
 app.use(helmet.permittedCrossDomainPolicies());
 app.use(helmet.referrerPolicy());
 app.use(helmet.xssFilter());
+app.use(helmet.originAgentCluster());
```
Comment (EvanHahn) [inline README.md:48]: Please remove this line because it's not enabled by default.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
...
+
+### Added
+
+- `helmet.originAgentCluster`: disabled by default, set `true` enables it
```
Comment (EvanHahn) [inline CHANGELOG.md:7]: Please change this to:

```suggestion
- `helmet.originAgentCluster`: a new middleware for the `Origin-Agent-Cluster` middleware, disabled by default
```

File: README.md [hunk 4]
```diff
@@ -479,3 +480,21 @@ app.use(helmet.xssFilter());
 You can install this module separately as `x-xss-protection`.
 
 </details>
+
+<details>
+<summary><code>helmet.originAgentCluster()</code></summary>
+
+`helmet.originAgentCluster` The Origin-Agent-Cluster header provides a mechanism to allow web applications to isolate their origins.
```
Comment (EvanHahn) [inline README.md:487]:
```suggestion
`helmet.originAgentCluster` sets the `Origin-Agent-Cluster` header, which provides a mechanism to allow web applications to isolate their origins. Read more about it [in the spec](https://whatpr.org/html/6214/origin.html#origin-keyed-agent-clusters).
```

File: index.ts [hunk 5]
```diff
@@ -78,11 +81,18 @@ const helmet: Helmet = Object.assign(
...
+    const arrayOfOptions = Object.values(options);
+    if (arrayOfOptions.some((option) => option === true)) {
+      if (
+        !(
+          options.originAgentCluster &&
+          arrayOfOptions.filter(Boolean).length === 1
+        )
+      ) {
+        throw new Error(
+          "Helmet no longer supports `true` as a middleware option, exception is Origin-Agent-Cluster. Remove the property from your options to fix this error."
```
Comment (EvanHahn) [inline index.ts:93]:
```suggestion
          "Helmet no longer supports `true` as a middleware option, except for Origin-Agent-Cluster. Remove the property from your options to fix this error."
```

**Review by EvanHahn — CHANGES_REQUESTED** [review]
Thank you very much for doing this! It looks great but needs a few small changes.

**Comment by salesh** [comment]
@EvanHahn
No Evan, thank you very much for giving me the opportunity to learn and contribute - you had written really full and descriptive guide, my part is really small. I had learned a lot about how a helmet works because of this. Thank you

**Comment by EvanHahn** [comment]
Thank you so much! I've made some small cleanups in #287 and will deploy soon.

**Comment by EvanHahn** [comment]
This has been published in `helmet@4.4.0`. Thanks again!

Feel free to add yourself to [the contributors list][0], or let me know your name and URL and I can do it.

[0]: https://github.com/helmetjs/helmetjs.github.io/blob/master/contributors/index.md

**Pull request merged**